src/commands/utils/acrs.ts

```diff
@@ -0,0 +1,66 @@
+import { Registry } from "@azure/arm-containerregistry";
+import { getAcrClient, getAcrManagementClient, listAll } from "./arm";
+import { Errorable, map as errmap, getErrorMessage } from "./errorable";
+import { ReadyAzureSessionProvider } from "../../auth/types";
+
+/**
+ * A registry with the id, name, location, and loginServer properties guaranteed to be defined.
+ */
+export type DefinedRegistry = Registry & Required<Pick<Registry, "id" | "name" | "location" | "loginServer">>;
+
+export async function getAcrs(
+    sessionProvider: ReadyAzureSessionProvider,
+    subscriptionId: string,
+    resourceGroup: string,
+): Promise<Errorable<DefinedRegistry[]>> {
+    const client = getAcrManagementClient(sessionProvider, subscriptionId);
+    const registriesResult = await listAll(client.registries.listByResourceGroup(resourceGroup));
+    return errmap(registriesResult, (registries) => registries.filter(asDefinedRegistry));
+}
+
+export async function getAcrRegistry(
+    sessionProvider: ReadyAzureSessionProvider,
+    subscriptionId: string,
+    resourceGroup: string,
+    acrName: string,
+): Promise<Errorable<DefinedRegistry>> {
+    const client = getAcrManagementClient(sessionProvider, subscriptionId);
+    try {
+        const registryResult = await client.registries.get(resourceGroup, acrName);
+        if (asDefinedRegistry(registryResult)) {
+            return { succeeded: true, result: registryResult };
+        }
+        return {
```

Comment:
There's a typo there: `asDefinedRegistry` is a [type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) and should be named `isDefinedRegistry`.

I'll fix that, which should make it clear that the `succeeded: false` inside the `try` block is happening because the registry is not properly defined (one of the mandatory properties is not set).